datacube_stats/utils/__init__.py

```diff
@@ -96,6 +96,53 @@ def to_float(da):
     return data.apply(to_float, keep_attrs=True)
 
 
+def cast_back(data, measurements):
```

Comment:
This should really do the following:

1. If requested dtype is different from supplied -- make a cast
ALSO
2. If supplied was floating point and output has nodata defined -- apply mask to output where input had nans

So if I give to this function int32 but expect int8, this should still work. For now at least let's not worry about nodata conversion for input types other than float(32|64). 

It's just a generalisation of numpy.cast to DataSet with extra handling of `nodata`, so let's not assume too much about the context it is used in.

Reply:
I think I got it right this time.